### PR TITLE
feat: post modal/page for welcome post type

### PR DESCRIPTION
**Pull request by sshanzel**
### Describe what this PR does
- The UI of welcome post and share post looks very very similar, understandably since because they are both a Squad post.
- Introduce new type of post which requires different rendering.
- Fixed the SharePost author image size (should have been 64px all along).
- Refactored the SquadPostModal to be reused for welcome post type.

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-WT-1329 #done

### CODE REVIEW

File: packages/shared/src/components/post/SharePostContent.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
+import React, { ReactElement, useContext, useMemo, useState } from 'react';
```
Comment (sshanzel) [inline packages/shared/src/components/post/SharePostContent.tsx:1]: Copy-pasted the whole current elements after the title.

File: packages/shared/src/components/ProfilePicture.tsx [hunk 2]
```diff
@@ -26,23 +27,25 @@ export interface ProfilePictureProps
...
-const sizeClasses = {
+const sizeClasses: Record<ProfileImageSize, string> = {
   xsmall: 'w-5 h-5',
   small: 'w-6 h-6',
   medium: 'w-8 h-8',
   large: 'w-10 h-10',
   xlarge: 'w-12 h-12',
   xxlarge: 'w-14 h-14',
-  xxxlarge: 'w-24 h-24',
+  xxxlarge: 'w-16 h-16',
```
Comment (sshanzel) [inline packages/shared/src/components/ProfilePicture.tsx:37]: Our guideline states `xxxlarge` to be `w-16 h-16`.

Comment (rebelchris) [reply packages/shared/src/components/ProfilePicture.tsx:37]: Does it have impact on any existing code?
Please tripple check 🙏

File: packages/shared/src/components/Feed.tsx [hunk 3]
```diff
@@ -133,6 +133,7 @@ const getStyle = (useList: boolean, spaciness: Spaciness): CSSProperties => {
 const PostModalMap: Record<PostType, typeof ArticlePostModal> = {
   [PostType.Article]: ArticlePostModal,
   [PostType.Share]: SharePostModal,
+  [PostType.Welcome]: SharePostModal,
```
Comment (rebelchris) [inline packages/shared/src/components/Feed.tsx:136]: Why do you add this if it's the same modal?
Not quite sure I get this one.

File: packages/shared/src/components/ProfilePicture.tsx [hunk 4]
```diff
@@ -11,7 +11,8 @@ export type ProfileImageSize =
   | 'large'
   | 'xlarge'
   | 'xxlarge'
-  | 'xxxlarge';
+  | 'xxxlarge'
+  | 'profile';
```
Comment (rebelchris) [inline packages/shared/src/components/ProfilePicture.tsx:15]: Rather check with Tomer, I'd like to keep things uniform

**Review by rebelchris — COMMENTED** [review]
Looks good, just not sure why we have a new modal when it's all the same?

File: packages/shared/src/components/ProfilePicture.tsx (hunk 2 above)
Comment (sshanzel) [reply packages/shared/src/components/ProfilePicture.tsx:37]: There was only one implementation which is the profile. Should be covered here 🚢

File: packages/shared/src/components/Feed.tsx (hunk 3 above)
Comment (sshanzel) [reply packages/shared/src/components/Feed.tsx:136]: This is to identify which modal the feed will use for the post clicked.

Comment (rebelchris) [reply packages/shared/src/components/Feed.tsx:136]: But they all render the same modal?
Should we not rahter have a PostModal with prop type? (It seems a bit redundant to have 2 calls do the same modal.

Comment (rebelchris) [reply packages/shared/src/components/Feed.tsx:136]: Ah sorry was misreading this piece, it's just the render that's the same.
For some reason thought you where introducing a new lazy modal with the same modal rendered 😅

File: packages/shared/src/components/ProfilePicture.tsx (hunk 4 above)
Comment (sshanzel) [reply packages/shared/src/components/ProfilePicture.tsx:15]: Agree with that. Awaiting response now 🚢

**Review by rebelchris — APPROVED** [review]
Nice work Lee 🙏

**Pull request merged**